src/pipelinedefinition.cpp

```
@@ -216,92 +216,300 @@ void PipelineDefinition::makeSubscriptions(ModelManager& manager) {
+        }
+        return StatusCode::OK;
+    }
 
-            if (connection.second.size() == 0) {
-                SPDLOG_ERROR("Validation of pipeline({}) definition failed. Missing dependency mapping for node: {}", this->pipelineName, node.nodeName);
-                return StatusCode::PIPELINE_DEFINITION_MISSING_DEPENDENCY_MAPPING;
+    Status markModelInputAsConnected(const std::string& name) {
+        // If currently validated node is of type DL model, mark its input as connected
+        // by erasing from previously gathered input set.
+        // If such input cannot be found in the map, it means we refer
+        // to non existing model input or we already connected it to some other data source which is invalid.
+        if (dependantModelInstance->getInputsInfo().count(name) == 0) {
+            SPDLOG_ERROR("Validation of pipeline({}) definition failed. Node:{} model:{} version:{} has no input with name:{}",
+                pipelineName,
+                dependantNodeInfo.nodeName,
+                dependantNodeInfo.modelName,
+                dependantNodeInfo.modelVersion.value_or(0),
+                name);
+            return StatusCode::PIPELINE_CONNECTION_TO_MISSING_MODEL_INPUT;
+        }
+        if (remainingUnconnectedDependantModelInputs.erase(name) == 0) {
+            SPDLOG_ERROR("Validation of pipeline({}) definition failed. Node:{} model:{} version:{} input name:{} is connected to more than one data source",
+                pipelineName,
+                dependantNodeInfo.nodeName,
+                dependantNodeInfo.modelName,
+                dependantNodeInfo.modelVersion.value_or(0),
+                name);
+            return StatusCode::PIPELINE_MODEL_INPUT_CONNECTED_TO_MULTIPLE_DATA_SOURCES;
+        }
+        return StatusCode::OK;
+    }
+
+    Status validateConnection(const NodeInfo& dependencyNodeInfo, const InputPairs& mapping) {
+        // At this point dependency node can only be either DL model node or entry node.
```

Hmm maybe throw an exception otherwise instead of comment?

---

This comment is to remember in next release of model ensemble to update this part of code.